app/models/spree/shipping_method_decorator.rb

```diff
@@ -49,14 +49,6 @@ def available_to_order_with_distributor_check?(order, display_on=nil)
   end
   alias_method_chain :available_to_order?, :distributor_check
 
-  def within_zone?(order)
-    if order.ship_address
-      zone && zone.include?(order.ship_address)
-    else
-      true # Shipping methods are available before we've selected an address
-    end
-  end
```

Comment:
Agree. We should take this as part of the spree upgrade. We might have more like this one.